### PR TITLE
ledger2beancount: 2.1 -> 2.5

### PR DESCRIPTION
**Pull request by bhipple**
Also, move to fetch src off the official `beancount` github org and update
dependencies. I've run this on my ledger file and it seems to work without issue.

###### Things done

- [X] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [X] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [X] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [X] Ensured that relevant documentation is up to date
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 113255` at e0fedb74 run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages failed to build:</summary>
  <ul>
    <li><a href="https://gist.github.com/7f16c89bd3194dc8b3ff0e811c88f93a">ledger</a></li></li>
    <li><a href="https://gist.github.com/7de0771badb0c845f862bb2cae7e7c7a">ledger-autosync</a></li></li>
  </ul>
</details>
<details>
  <summary>1 package built:</summary>
  <ul>
    <li><a href="https://gist.github.com/132e751a3c65b890ff50d37edba3a285">ledger2beancount</a></li></li>
  </ul>
</details>
<details>
  <summary>1 suggestion:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-phase-hooks.md">missing-phase-hooks</a>

`installPhase` should probably contain `runHook preInstall` and `runHook postInstall`.

Near pkgs/tools/text/ledger2beancount/default.nix:41:3:
```
   |
41 |   installPhase = ''
   |   ^
```
</li>
  </ul>
</details>

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 113255` at e0fedb74 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages failed to build:</summary>
  <ul>
    <li><a href="https://gist.github.com/9b94e184121a3464e656e3300030d89f">ledger</a></li></li>
    <li><a href="https://gist.github.com/3884c96434d7348266b8acba090dbecd">ledger-autosync</a></li></li>
  </ul>
</details>
<details>
  <summary>1 package built:</summary>
  <ul>
    <li><a href="https://gist.github.com/609bf94e067256bdda828615a4c08452">ledger2beancount</a></li></li>
  </ul>
</details>
<details>
  <summary>1 suggestion:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-phase-hooks.md">missing-phase-hooks</a>

`installPhase` should probably contain `runHook preInstall` and `runHook postInstall`.

Near pkgs/tools/text/ledger2beancount/default.nix:41:3:
```
   |
41 |   installPhase = ''
   |   ^
```
</li>
  </ul>
</details>

File: pkgs/applications/office/ledger/default.nix [hunk 1]
```diff
@@ -27,11 +25,12 @@ stdenv.mkDerivation rec {
...
   ];
 
-  # by default, it will query the python interpreter for it's sitepackages location
-  # however, that would write to a different nixstore path, pass our own sitePackages location
+  # By default, it will query the python interpreter for its sitepackages location.
+  # However, that would write to a different nixstore path; pass our own sitePackages location
   prePatch = lib.optionalString usePython ''
-    substituteInPlace src/CMakeLists.txt \
-      --replace 'DESTINATION ''${Python_SITEARCH}' 'DESTINATION "${python.sitePackages}"'
+    echo substituteInPlace src/CMakeLists.txt \
```
Comment (SuperSandro2000) [inline pkgs/applications/office/ledger/default.nix:31]:
```suggestion
    substituteInPlace src/CMakeLists.txt \
```

File: pkgs/applications/office/ledger/default.nix [hunk 2]
```diff
@@ -12,11 +12,9 @@ stdenv.mkDerivation rec {
     sha256 = "0x6jxwss3wwzbzlwmnwb8yzjk8f9wfawif4f1b74z2qg6hc4r7f6";
   };
 
-  outputs = [ "out" "dev" ];
```
Comment (SuperSandro2000) [inline pkgs/applications/office/ledger/default.nix:15]: Why did you remove this?

Comment (bhipple) [reply pkgs/applications/office/ledger/default.nix:15]: The changes to ledger were in my shadow and I didn't mean to add them to this PR; apologies!  Will update.

(The `dev` output happens to be both broken and also unused, though, which is why I was looking at patching/fixing/changing it somehow.)

Comment (SuperSandro2000) [reply pkgs/applications/office/ledger/default.nix]: The output was really empty? Then it can be removed.
